index.js

```diff
@@ -1,187 +1,210 @@
-      owner: 'cds-snc',
-      repo: 'notification-terraform',
-      path: '.github/workflows/merge_to_main_production.yml'
-    });
+  const { data: data } = await octokit.repos.getContents({
+    owner: "cds-snc",
+    repo: "notification-terraform",
+    path: ".github/workflows/merge_to_main_production.yml",
+  });
 
-    const fileContent = Base64.decode(data.content);
-    const prodVersion = fileContent.match(/INFRASTRUCTURE_VERSION: '(.*)'/)[1];
+  const fileContent = Base64.decode(data.content);
+  const prodVersion = fileContent.match(/INFRASTRUCTURE_VERSION: '(.*)'/)[1];
 
-    const latestVersion = (await getLatestTag('notification-terraform')).replace('v', '');
+  const latestVersion = (await getLatestTag("notification-terraform")).replace(
+    "v",
+    ""
+  );
 
-    return prodVersion != latestVersion;
+  return prodVersion != latestVersion;
 }
 
 async function run() {
-    const { data: data } = await octokit.repos.getContents({
-      owner: 'cds-snc',
-      repo: 'notification-manifests',
-      path: 'env/production/kustomization.yaml'
-    });
+  const { data: data } = await octokit.repos.getContents({
+    owner: "cds-snc",
+    repo: "notification-manifests",
+    path: "env/production/kustomization.yaml",
+  });
 
-    const { data: issueData } = await octokit.repos.getContents({
-      owner: 'cds-snc',
-      repo: 'notification-manifests',
-      path: '.github/PULL_REQUEST_TEMPLATE.md'
-    });
+  const { data: issueData } = await octokit.repos.getContents({
+    owner: "cds-snc",
+    repo: "notification-manifests",
+    path: ".github/PULL_REQUEST_TEMPLATE.md",
+  });
 
-    const apiSha = await getHeadSha("notification-api");
-    const adminSha = await getHeadSha("notification-admin");
-    const manifestsSha = await getHeadSha("notification-manifests", "main");
-
-    fileContent = YAML.parse(Base64.decode(data.content));
-    issueContent = Base64.decode(issueData.content);
-
-    fileContent.images.forEach((image) => {
-      if(image.name == "admin"){
-        oldAdminSha = image.newName.split(":").slice(-1)[0]
-        image.newName = `${AWS_ECR_URL}/notify-admin:${adminSha.slice(0,7)}`
-      }
-      if(image.name == "api"){
-        oldApiSha= image.newName.split(":").slice(-1)[0]
-        image.newName = `${AWS_ECR_URL}/notify-api:${apiSha.slice(0,7)}`
-      }
-    });
+  const apiSha = await getHeadSha("notification-api");
+  const adminSha = await getHeadSha("notification-admin");
+  const manifestsSha = await getHeadSha("notification-manifests", "main");
 
-    const newBlob = Base64.encode(YAML.stringify(fileContent))
+  const fileContent = YAML.parse(Base64.decode(data.content));
+  const issueContent = Base64.decode(issueData.content);
 
-   if(newBlob !== data.content){
-      closePRs()
+  fileContent.images.forEach((image) => {
+    if (image.name == "admin") {
+      oldAdminSha = image.newName.split(":").slice(-1)[0]; // eslint-disable-line no-unused-vars, no-undef
```

Comment:
They're actually used later, but IIRC (last Friday is far..), the linter performs the wrong analysis and only check the scope where the variables *should* be used. These are declared globally here and can be reused later on. They are defined as a side-effect of the `forEach` loop, which is also a red flag. I changed this in another branch but not here, as I wanted to avoid logic changes of any sort.